client/clientutil/snapinfo_test.go

```diff
@@ -0,0 +1,278 @@
+		Website:   "http://example.com/thingy",
+		StoreURL:  "https://snapcraft.io/thingy",
+		Broken:    "broken",
+	}
+	// valid InstallDate
+	err := os.MkdirAll(si.MountDir(), 0755)
+	c.Assert(err, IsNil)
+	err = os.Symlink(si.Revision.String(), filepath.Join(filepath.Dir(si.MountDir()), "current"))
+	c.Assert(err, IsNil)
+
+	ci, err := clientutil.ClientSnapFromSnapInfo(si, nil)
+	c.Check(err, IsNil)
+
+	// check that fields are filled
+	// see daemon/snap.go for fields filled after this
+	expectedZeroFields := []string{
+		"Screenshots", // unused nowadays
+		"DownloadSize",
+		"InstalledSize",
+		"Health",
+		"Status",
+		"TrackingChannel",
+		"IgnoreValidation",
+		"CohortKey",
+		"DevMode",
+		"TryMode",
+		"JailMode",
+		"MountedFrom",
+	}
+	var checker func(string, reflect.Value)
+	checker = func(pfx string, x reflect.Value) {
+		t := x.Type()
+		for i := 0; i < x.NumField(); i++ {
+			f := t.Field(i)
+			if f.PkgPath != "" {
+				// not exported, ignore
+				continue
+			}
+			v := x.Field(i)
+			if f.Anonymous {
+				checker(pfx+f.Name+".", v)
+				continue
+			}
+			if reflect.DeepEqual(v.Interface(), reflect.Zero(f.Type).Interface()) {
+				name := pfx + f.Name
+				c.Check(expectedZeroFields, testutil.Contains, name, Commentf("%s not set", name))
+			}
+		}
+	}
+	x := reflect.ValueOf(ci).Elem()
+	checker("", x)
+
+	// check some values
+	c.Check(ci.Name, Equals, "the-snap_insta")
+	c.Check(ci.Type, Equals, "app")
+	c.Check(ci.ID, Equals, si.ID())
+	c.Check(ci.Revision, Equals, snap.R(99))
+	c.Check(ci.Version, Equals, "v1")
+	c.Check(ci.Title, Equals, "the-title")
+	c.Check(ci.Summary, Equals, "the-summary")
+	c.Check(ci.Description, Equals, "the-description")
+	c.Check(ci.Icon, Equals, si.Media.IconURL())
+	c.Check(ci.Website, Equals, si.Website)
+	c.Check(ci.StoreURL, Equals, si.StoreURL)
+	c.Check(ci.Developer, Equals, "thingyinc")
+	c.Check(ci.Publisher, DeepEquals, &si.Publisher)
+}
+
+type testStatusDecorator struct{}
+
+func (sd *testStatusDecorator) DecorateWithStatus(appInfo *client.AppInfo, app *snap.AppInfo) error {
+	if appInfo.Snap != app.Snap.InstanceName() || appInfo.Name != app.Name {
+		panic("mismatched")
+	}
+	appInfo.Enabled = true
+	appInfo.Active = true
+	return nil
+}
+
+func (*cmdSuite) TestClientSnapFromSnapInfoAppsInactive(c *C) {
+	si := &snap.Info{
+		SnapType:      snap.TypeApp,
+		SuggestedName: "",
+		InstanceKey:   "insta",
+		SideInfo: snap.SideInfo{
+			RealName: "the-snap",
+			SnapID:   "snapidid",
+			Revision: snap.R(99),
+		},
+	}
+	si.Apps = map[string]*snap.AppInfo{
+		"svc": {Snap: si, Name: "svc", Daemon: "simple"},
+		"app": {Snap: si, Name: "app", CommonID: "common.id"},
+	}
+	// sanity
+	c.Check(si.IsActive(), Equals, false)
+	// desktop file
+	df := si.Apps["app"].DesktopFile()
+	err := os.MkdirAll(filepath.Dir(df), 0755)
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(df, nil, 0644)
+	c.Assert(err, IsNil)
+
+	ci, err := clientutil.ClientSnapFromSnapInfo(si, &testStatusDecorator{})
```

Comment:
I initially stumbled here and did not remember that the testStatusDecorator is only called for active apps. If we keep this, maybe we can add a small comment below like: "// decorator function not called because app not active" or similar?